src/units.jl

```diff
@@ -0,0 +1,106 @@
+module Units
+
+import ..DEFAULT_DIM_TYPE
+import ..DEFAULT_VALUE_TYPE
+import ..Quantity
+
+@assert DEFAULT_VALUE_TYPE == Float64 "`units.jl` must be updated to support a different default value type."
+
+macro add_prefixes(base_unit, prefixes)
+    @assert prefixes.head == :tuple
+    expr = _add_prefixes(base_unit, prefixes.args)
+    return expr |> esc
+end
+
+function _add_prefixes(base_unit::Symbol, prefixes)
+    all_prefixes = (
+        f=1e-15, p=1e-12, n=1e-9, μ=1e-6, u=1e-6, m=1e-3, c=1e-2,
+        k=1e3, M=1e6, G=1e9, T=1e12, P=1e15
+    )
+    expr = Expr(:block)
+    for (prefix, value) in zip(keys(all_prefixes), values(all_prefixes))
+        prefix in prefixes || continue
+        new_unit = Symbol(prefix, base_unit)
+        push!(expr.args, :(const $new_unit = $value * $base_unit))
+    end
+    return expr
+end
+
+# SI base units
+const m = Quantity(1.0, length=1)
+const g = Quantity(1e-3, mass=1)
+const s = Quantity(1.0, time=1)
+const A = Quantity(1.0, current=1)
+const K = Quantity(1.0, temperature=1)
+const cd = Quantity(1.0, luminosity=1)
+const mol = Quantity(1.0, amount=1)
+
+@add_prefixes m (f, p, n, μ, u, c, m, k, M, G)
+@add_prefixes g (μ, u, m, k)
+@add_prefixes s (f, p, n, μ, u, m)
+@add_prefixes A (n, μ, u, m, k)
+@add_prefixes K (m,)
+@add_prefixes cd (m,)
+@add_prefixes mol (m,)
+
+# SI derived units
+const Hz = inv(s)
+const N = kg * m / s^2
+const Pa = N / m^2
+const J = N * m
+const W = J / s
+const C = A * s
+const V = W / A
+const F = C / V
+const Ω = V / A
+const T = N / (A * m)
+
+@add_prefixes Hz (k, M, G)
+@add_prefixes N ()
+@add_prefixes P ()
+@add_prefixes J ()
+@add_prefixes W (k, M, G)
+@add_prefixes C ()
+@add_prefixes V (m, k, M, G)
+@add_prefixes F ()
+@add_prefixes Ω (m,)
+@add_prefixes T ()
+
+# Assorted units
+const min = 60 * s
+const h = 60 * min
+const day = 24 * h
+const wk = 7 * day
+const yr = 365.25 * day
```

Comment:
This seems wrong. I don't know what other measurement systems do, but this is a pretty steep approximation.

Reply:
This is just the SI definition of a year. The SI definition is different from a literal orbit around the sun (which has measurement uncertainty associated with it).